db/inlineskiplist_test.cc

```diff
@@ -42,7 +43,44 @@ struct TestComparator {
   }
 };
 
-class InlineSkipTest : public testing::Test {};
+typedef InlineSkipList<TestComparator> TestInlineSkipList;
+
+class InlineSkipTest : public testing::Test {
+ public:
+  void Insert(TestInlineSkipList& list, Key key) {
+    char* buf = list.AllocateKey(sizeof(Key));
+    memcpy(buf, &key, sizeof(Key));
+    list.Insert(buf);
+    keys_.insert(key);
+  }
+
+  void InsertWithHint(TestInlineSkipList& list, Key key,
+                      TestInlineSkipList::InsertHint** hint) {
+    char* buf = list.AllocateKey(sizeof(Key));
+    memcpy(buf, &key, sizeof(Key));
+    list.InsertWithHint(buf, hint);
+    keys_.insert(key);
+  }
+
+  void Validate(TestInlineSkipList& list) {
+    for (Key key : keys_) {
+      ASSERT_TRUE(list.Contains(Encode(&key)));
+    }
+    TestInlineSkipList::Iterator iter(&list);
+    ASSERT_FALSE(iter.Valid());
+    Key zero = 0;
+    iter.Seek(Encode(&zero));
+    for (Key key : keys_) {
+      ASSERT_TRUE(iter.Valid());
+      ASSERT_EQ(key, Decode(iter.key()));
+      iter.Next();
+    }
+    ASSERT_FALSE(iter.Valid());
+  }
```

Comment:
This validation only validates the link of the next level is correct.

To validate it is a valid skip list, we also need to validate every level is at the correct order, for every node for a higher level ink, the node is included in lower level too.